app/controllers/saml_idp_controller.rb

```diff
@@ -227,4 +229,22 @@ def resolved_authn_context_int_ial
   def require_path_year
     render_not_found if params[:path_year].blank?
   end
+
+  def unknown_authn_contexts
+    return nil if saml_request.requested_vtr_authn_contexts.present?
+    return nil if requested_authn_contexts.blank?
+
+    (requested_authn_contexts -
+      Saml::Idp::Constants::VALID_AUTHN_CONTEXTS).reject do |authn_context|
+      authn_context.match(req_attrs_regexp)
+    end.join(' ').presence
```

Comment:
indentation is kinda funky here, WDYT of breaking this out into a variable?
```suggestion
    unmatched_contexts = Saml::Idp::Constants::VALID_AUTHN_CONTEXTS).reject do |authn_context|
      authn_context.match(req_attrs_regexp)
    end
    (requested_authn_contexts - unmatched_contexts).join(' ').presence
```

Reply:
sure! updated in https://github.com/18F/identity-idp/pull/11362/commits/52a697bc26fb39fc2d57df7f93f6808bb228029a (i pulled it into a separate method to make it easier to read)